tools/delete_imaging_upload.pl

```diff
@@ -0,0 +1,1712 @@
+    ['-ignore'   , 'const'        , 0, \$options{'DIE_ON_FILE_ERROR'}, 
+     'Ignore files that exist in the database but not on the file system.'
+     . ' Default is to abort if such a file is found.'],
+    ['-nofilesbk', 'const'        , 1, \$options{'NO_FILES_BK'},
+     'Do not backup the files produced by the MRI pipeline. Default is to backup all files '
+     . 'to be deleted'],
+    ['-uploadID' , 'string'       , 1, \$options{'UPLOAD_ID'},
+     'Comma-separated list of upload IDs to delete. All the uploads must be associated to the same archive.'],
+    ['-protocol' , 'const'        , 1, \$options{'DELETE_PROTOCOLS'},
+     'Delete the entries in mri_processing_protocols tied to the upload(s) specified via -uploadID.'],
+    ['-form'     , 'const'        , 1, \$options{'DELETE_MRI_PARAMETER_FORM'},
+     'Delete the entries in mri_parameter_form tied to the upload(s) specified via -uploadID.'],
+    ['-type'     , 'string'       , 1, \$scanTypeList, 'Comma-separated list of scan types to delete.'],
+    ['-nosqlbk'  , 'const'        , 1, \$options{'NO_SQL_BK'},
+     'Do not backup the information deleted from the database with mysqldump. '
+      . 'Default is to backup all records deleted in a file named backup.sql'],
+    ['-defaced'  , 'const'        , 1, \$options{'KEEP_DEFACED'},
+     'Replace each MINC files whose scan types are in the list of types to deface with its'
+      . ' corresponding defaced file.']
+);
+
+my $Help = <<HELP;
+HELP
+
+my $usage = <<USAGE;
+Usage: $0 [-profile file] [-ignore] [-backup basename] [-protocol] [-form] [-uploadID list_of_uploadIDs]
+            [-type list_of_scan_types] [-defaced] [-nosqlbk] [-nofilesbk]
+USAGE
+
+&Getopt::Tabular::SetHelp($Help, $usage);
+&Getopt::Tabular::GetOptions(\@opt_table, \@ARGV) || exit $NeuroDB::ExitCodes::GETOPT_FAILURE;
+
+if(@ARGV != 0) {
+    print STDERR "$usage\n";
+    exit $NeuroDB::ExitCodes::INVALID_ARG;
+}
+
+if(defined $scanTypeList && $options{'DELETE_MRI_PARAMETER_FORM'}) {
+    print STDERR "Option -type cannot be used in conjunction with option -form. Aborting.\n";
+       exit $NeuroDB::ExitCodes::INVALID_ARG;
+} 
+
+if(defined $scanTypeList && $options{'KEEP_DEFACED'}) {
+    print STDERR "Option -type cannot be used in conjunction with option -defaced. Aborting.\n";
+       exit $NeuroDB::ExitCodes::INVALID_ARG;
+} 
+
+if($options{'KEEP_DEFACED'} && $options{'DELETE_MRI_PARAMETER_FORM'}) {
+    print STDERR "Option -defaced cannot be used in conjunction with option -form. Aborting.\n";
+       exit $NeuroDB::ExitCodes::INVALID_ARG;
+} 
+
+if($options{'UPLOAD_ID'} eq '') {
+    print STDERR "Missing -uploadID option. Aborting.\n";
+    exit $NeuroDB::ExitCodes::MISSING_ARG;
+}
+
+if($options{'BACKUP_BASENAME'} ne '' && $options{'NO_FILES_BK'} && $options{'NO_SQL_BK'}) {
+    print STDERR "Option -backup cannot be used if -nofilesbk and -nosqlbk are also used. Aborting.\n";
+    exit $NeuroDB::ExitCodes::INVALID_ARG;
+}
```

Comment:
Could you add a check here to make sure that the file `$options{'BACKUP_BASENAME'}` does not already exist? Just to make sure that the user does not overwrite a backup by mistake.

Reply:
Ok. The script will abort if the backup already exists.